dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/configuration/UploadSchedulerStrategy.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.configuration
+
+/**
+ * Defines the strategy used to schedule the waiting period between batch uploads.
+ */
+interface UploadSchedulerStrategy {
+
+    /**
+     * Should return the delay in millisecond to wait until the next upload attempt
+     * is performed.
+     * @param featureName the name of the feature for which a new upload will be scheduled
+     * @param uploadAttempts the number of requests that were attempted during this run. Will be zero if the device
+     * is not ready (e.g.: when offline or with low battery). If multiple batches can be uploaded, the attempts will
+     * stop at the first failure.
+     * @param lastStatusCode the HTTP status code of the last request (if available). A successful upload will have a
+     * status code 202 (Accepted). When null, it means that
```

Comment:
This will be exposed, but we won't document it in the website, so almost no one will implement it. Also it is mentioned in the documentation that they should check for a 202.